### PR TITLE
Add Google Kubernetes Engine cluster setup

### PR DESCRIPTION
**Pull request by ottolote**
First contribution, any and all feedback is appreciated

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
**Commands should work for all environments but I will change to codetabs if that is preferred still**
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Add setup section for Google Kubernetes Engine and troubleshooting for GKE private clusters.

Main motivation for this PR is to document extra steps required for sidecar injection to work in private GKE clusters.

### CODE REVIEW

**Comment by yaron2** [comment]
Thanks for this!

File: daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-gke.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+type: docs
+title: "Setup a Google Kubernetes Engine cluster"
+linkTitle: "Google Kubernetes Engine"
+weight: 2000
```
Comment (orizohar) [inline daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-gke.md]: Looks like another page in this folder has the value "2000". This is used to order the articles in the directory so these should have different values. Please change this value to "3000"

**Review by orizohar — CHANGES_REQUESTED** [review]
This is a great addition to the docs! Thank You! 
Well written, just a tiny comment on the Hugo metadata

File: daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-gke.md (hunk 1 above)
Comment (ottolote) [reply daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-gke.md]: Changed to 3000 :+1:

I see both `setup-aks.md` and `setup-minikube.md` in this folder have a weight of 2000 so I took the liberty of changing the weight of the minikube page to 1000

Comment (orizohar) [reply daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-gke.md]: Thanks for catching that and fixing :)

**Review by orizohar — APPROVED** [review]

**Pull request merged**